server/channels/app/scheduled_post_job.go

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+	"github.com/pkg/errors"
+)
+
+const (
+	getPendingScheduledPostsPageSize = 10
+	scheduledPostBatchWaitTime       = 1 * time.Second
+)
+
+func (a *App) ProcessScheduledPosts(rctx request.CTX) {
+	rctx = rctx.WithLogger(rctx.Logger().With(mlog.String("component", "scheduled_post_job")))
+	rctx.Logger().Debug("ProcessScheduledPosts called...")
+
+	beforeTime := model.GetMillis()
+	lastScheduledPostId := ""
+
+	for {
```

Comment:
Is there a reason this job isn't hooked into the job server? Did we decide that it was better not to for a certain reason? I guess that's why we aren't using the `BatchWorker` here.

Reply:
With job server we don't have control over when the processes starts. With scheduled posts we need to be aligned to a clock time, as in run at every 5 mins, for example, 12:05, 12:10, 12:15 etc. to ensure timely delivery of scheduled posts. This is not achieved by batch/job server. its good for stuff thats doesn't have a delivery time. LMK if I'm not understanding it correctly.